### ISSUE TITLE
ADDSynth: Allow for bigger frequency range

### ISSUE THREAD

**Issue by badosu** [zynaddsubfx/zyn-fusion-issues#189]
Hello, I am trying to design a kick using zynaddsubfx and I found a limitation in the process.

It is related to the frequency envelope in which the upper limit can't be expanded. I am using this video as a reference: https://www.youtube.com/watch?v=gJOdL6R4olw

Zynaddsubfx allows me to reach the sound done from 0:26 to 0:34, which I do by setting up the frequency envelope, however I can't expand the frequency range as shown between 1:03 an 1:08.

This is the 'de facto' method to generate a certain kick sound that is fundamental producing for class of EDM genres.

Let me know what you think of this feature, if simple enough I can try to take a stab at it.

Thanks!

**Comment by fundamental** [comment]
Yeah, I could see this issue making sense. Zyn does have a good number of issues with large swings in frequency due to the current antialiasing approach, though that's another issue to consider.

If you're implementing this, then I think the design makes the most sense would be the addition of an envelope scale parameter added to the EnvelopeParams and Envelope classes and then a UI control for the scale term getting exposed for all envelope types except amplitude envelopes (I think that makes sense?). Exposing the larger range might also lead you to bump into a few approximations along the way, but those have been on the "need to fix" list for years now.

**Comment by badosu** [comment]
I was thinking in using a port for the frequency envelope range as an integer in the range of 2-100 semitones.

However if the Envelope being used for the frequency uses a generic class it indeed makes sense to make it a new scale parameter for any envelope. Would be a bit of an overkill since I only see an application for this at the moment, and frontend implementation, but doable.

**Comment by fundamental** [comment]
I'd say a happy middleground would be implementing the backend such that the scale factor is in the generic Envelope class, but only worrying about the frontend implementation for frequency envelopes.
That way it's easy for future work to expand the other envelopes while keeping the required work for the feature you want to a lower level.
